### PR TITLE
Find replace toolbar

### PR DESCRIPTION
**Pull request by fitzk**
## Summary by CodeRabbit

- **New Features**
	- Improved tab management for better user experience when the extension is activated.
	- Enhanced iframe management and event handling for demos, improving responsiveness and clarity.
	- Introduced structured event enums for better handling of application events.
	- Added functions for streamlined icon management based on application events.
	- Enabled source maps for JavaScript files in development mode to enhance debugging capabilities.

- **Bug Fixes**
	- Streamlined icon management and event handling logic for clarity and maintainability.

- **Documentation**
	- Added enumerations for various application events to improve clarity in event handling.

### CODE REVIEW

File: apps/mocksi-lite-next/src/pages/background/index.ts [hunk 1]
```diff
@@ -118,68 +146,69 @@ chrome.runtime.onMessage.addListener(
...
   (request, _sender, sendResponse) => {
-    // This logging is useful and only shows up in the service worker
-    console.log(" ");
-    console.log("Previous message from external:", prevRequest);
-    console.log("Received new message from external:", request);
+    console.log(request);
 
     // execute in async block so that we return true
     // synchronously, telling chrome to wait for the response
     (async () => {
```
Comment (elg0nz) [inline apps/mocksi-lite-next/src/pages/background/index.ts:153]: I really want us to refactor this block, see the changes on https://github.com/Mocksi/HARlighter/pull/184

File: apps/mocksi-lite-next/src/pages/background/index.ts [hunk 2]
```diff
@@ -43,10 +43,24 @@ const clearAuth = async (): Promise<void> => {
 };
 
 async function getCurrentTab() {
```
Comment (elg0nz) [inline apps/mocksi-lite-next/src/pages/background/index.ts:45]: could you explain a little these changes?

File: apps/mocksi-lite-next/src/pages/background/index.ts [hunk 3]
```diff
@@ -118,68 +146,69 @@ chrome.runtime.onMessage.addListener(
...
           });
         }
       } else {
         const tab = await getCurrentTab();
         if (!tab?.id) {
-          sendResponse({ message: request.message, status: "no-tab" });
-          console.log("No active tab found, could not send message");
+          sendResponse({
+            message: request.message,
+            status: ExtHarnessEvents.NO_TAB,
```
Comment (elg0nz) [inline apps/mocksi-lite-next/src/pages/background/index.ts:181]: Just curious what `ExtHarnessEvents` means?

File: apps/mocksi-lite-next/src/pages/events.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  RETRY = "RETRY",
+}
+
+export enum DemoEditEvents {
+  NEW_EDIT = "NEW_EDIT",
+  CHAT_MESSAGE = "CHAT_MESSAGE",
+  CHAT_RESPONSE = "CHAT_RESPONSE",
+}
+
+export enum ExtHarnessEvents {
```
Comment (elg0nz) [inline apps/mocksi-lite-next/src/pages/events.ts:21]: Not a fan of the name `ExtHarness` could something like Layout work better?

**Review by elg0nz — CHANGES_REQUESTED** [review]
This is heading in a great direction! I’m excited to see the progress 😄  That said, I’d recommend a few tweaks to improve readability and make the code easier to maintain in the long run. Let me know what you think!

File: apps/mocksi-lite-next/src/pages/events.ts (hunk 4 above)
Comment (fitzk) [reply apps/mocksi-lite-next/src/pages/events.ts:21]: I means Extension Harness haha I have know idea why I call the iframe that when using this pattern but I have for so long I didn't even think about what a mouthful that is haha I'll change it to something shorter

File: apps/mocksi-lite-next/src/pages/background/index.ts (hunk 2 above)
Comment (fitzk) [reply apps/mocksi-lite-next/src/pages/background/index.ts:45]: yeah so for a getCurrentTab, windowType: "normal",  limits tabs we query to the main browser window (as opposed to the dev tools, or a side panel window) so we won't be querying  dev tools or a side panel. 

I also added a fallbackTab variable which is set when the user opens the extension via clicking on the toolbar icon, the action returns the tab that the extension is mounted in

File: apps/mocksi-lite-next/src/pages/content/mocksi-extension.tsx [hunk 5]
```diff
@@ -155,56 +152,94 @@ chrome.runtime.onMessage.addListener((request) => {
...
+                    }
+                    await startDemo(request);
+                  }
                 }
-                reactor.attach(document, getHighlighter());
               }
-              if (request.message === "NEW_EDIT") {
+
+              if (
+                request.message === AppEvents.EDIT_DEMO_STOP ||
```
Comment (jonathankap) [inline apps/mocksi-lite-next/src/pages/content/mocksi-extension.tsx:196]: Nit: don't you already have this in requestingStopDemo?

**Review by jonathankap — APPROVED** [review]
Looks good

File: apps/mocksi-lite-next/src/pages/background/index.ts (hunk 1 above)
Comment (fitzk) [reply apps/mocksi-lite-next/src/pages/background/index.ts:145]: I'm going to merge this then look at incorporating your changes in a separate PR

File: apps/mocksi-lite-next/src/pages/content/mocksi-extension.tsx (hunk 5 above)
Comment (fitzk) [reply apps/mocksi-lite-next/src/pages/content/mocksi-extension.tsx]: yes fixed

**Pull request merged**